### DISCUSSION TITLE
Anyone out there?

### DISCUSSION THREAD

**Discussion by jareff-g** [jareff-g/ALT-Scann8#10] [Announcements]
Hello all,

I created this repository one year ago mainly to keep my T-Scann8 alternate software safe, for myself.

Now I see in the traffic report from Github that there appears to be more people using it. 

If you are real people and not some kind of bot, please manifest yourselves; If I know for sure there is more people using this, I can react accordingly (mainly being more careful with what I deliver in master).  Also would be  good to have some external feedback, and propositions for enhancements.

Cheers,
Juan

**Comment by ToRa-51** [comment]
Hello Juan, 

I am Ton Raps from Geleen in the south of the Netherlands and so far the only person, ToRa-51 (51 is my year of birth), who has made a fork of your repository for my own account on Github. 
By searching for keyword 'T-Scann8' on Google I found the site (moevi.nl) of Hans Meulenbroek from Borne in the Netherlands with an extensive report on his construction of the T-Scann8 scanner by Torulf Holström. It also mentions your name because of the hardware and software improvements. As a result, I ended up on your Github account and after some doubts I started installing the software and also making the hardware adjustments. The result made me feel good, but unfortunately I didn't get any further than the change you uploaded on October 14. Maybe done something wrong? Again on a new SD card everything reinstalled, but unfortunately gave the same error messages as the previous installation. 
I will create a new discussion about this. 

Greetings, Ton (ToRa-51).

![IMG_7560](https://github.com/jareff-g/ALT-Scann8/assets/26844127/14517aa5-062c-4e7d-bb88-1186adfac028)

![IMG_7564](https://github.com/jareff-g/ALT-Scann8/assets/26844127/e16818da-7319-43d1-a354-24efb38dc9d7)

**Reply by jareff-g** [reply]
Hello Ton,

Nice to hear from you. Happy to see ALT-Scann8 is used elsewhere. I was in touch with Hans a while ago, after he completed his T-Scann 8 scanner, but now I think he is busy with other formats (as probably you have seen it at his site).

So, regarding your issues, please fell free to open a new discussion with the details, including the log file, with ALT-Scann8 in debug mode. I see I made quite a few changes that day, but the cleanup I did on the I2C code is the primary suspect (although I do not see what could go wrong, works fine for me).

I see you are not using Torulf PCB, but a prototype board instead. Have you done any relevant changes with respect to Torulf design? The heath-sink in the stepper motor controllers look different, is it the same controller?

So, waiting for your report to check what could be wrong.

Regards,
Juan

PS: Nice cable routing by the way! I'll try to do something similar in the second scanner I'm building.

**Reply by ToRa-51** [reply]
Juan, 

Thanks for replying to my message.

> Have you done any relevant changes with respect to Torulf design? The heath-sink in the stepper motor controllers look different, is it the same controller?

I didn't make any special adjustments to the schematic except adding a 12V 6A power supply and a few buck converters (1 for the RPi and 1 for the rest of the electronics). 
For the 3W Power LED I only applied a dimmer. Might be useful. 
The heat-sink on the step motor controllers (A4988) are only slightly larger and therefore cool a bit better.

> PS: Nice cable routing by the way! I'll try to do something similar in the second scanner I'm building.

Thanks for the compliment. Wiring of units with electronics I have done as a daily work for almost 38 years. Just like many other jobs that an experienced electronics technician pretty much does.

PS: How do I create a log file with ALT-Scann8 in debug mode? 

Greetings, 
Ton.

**Reply by jareff-g** [reply]
Hello Ton,

I didn't thought about adding a second buck converter, probably that's why i was having frequent issues to start the scanner. Finally I decided to use a 12v to USB C converter (like [this](https://fr.aliexpress.com/item/1005005889810067.html?spm=a2g0o.order_list.order_list_main.5.11595e5b14HRSN&gatewayAdapt=glo2fra)) and now it start fine every time.

As for the log file, that's easy, just start ALT-Scann8 in the commend line as usual adding '-l DEBUG- at the end. More generally, you can also use '-h' in th ecommend line to get a list of all supported parameter. So, once started with log enabled, when problem happens, you can exit from ALT-Scann8 and send me the generated log file (in the same folder as the ALT-Scann8 python file, name is ALT-Scann8.YYYYMMDD.log, with the current date replacing the placeholders). Also a brief description of what the problem is, although from what you say I understand the problem happens on startup.

I'll have a look at the log file contents, and hopefully I'll be able to determine what the problem is, otherwise I'll add more logs in the suspected area and we try again.

Let me know.
Regards
Juan

**Comment by HansMeulenbroek** [comment]
Hello Juan, Hello Ton,

I’ve seen the photo’s of Ton’s build, very impressive. Just to let you know that after moving to North-Germany I finally got sufficient space for my digitizing hobby and have decided to try to install the new version of Juan’s software. To be continued!

Best regards,
Hans

Op 21 nov. 2023, om 19:07 heeft Juan ***@***.***> het volgende geschreven:


Hello Ton,

Nice to hear from you. Happy to see ALT-Scann8 is used elsewhere. I was in touch with Hans a while ago, after he completed his T-Scann 8 scanner, but now I think he is busy with other formats (as probably you have seen it at his site).

So, regarding your issues, please fell free to open a new discussion with the details, including the log file, with ALT-Scann8 in debug mode. I see I made quite a few changes that day, but the cleanup I did on the I2C code is the primary suspect (although I do not see what could go wrong, works fine for me).

I see you are not using Torulf PCB, but a prototype board instead. Have you done any relevant changes with respect to Torulf design? The heath-sink in the stepper motor controllers look different, is it the same controller?

So, waiting for your report to check what could be wrong.

Regards,
Juan

PS: Nice cable routing by the way! I'll try to do something similar in the second scanner I'm building.

**Reply by jareff-g** [reply]
Hello Hans,

Glad to see you are back. 

Let's see if some of the anonymous users who regularly clone the ALT-Scann8 and AfterScan repositories are encouraged to open a Github account and join the group. It will help for future developments to have feedback or, better yet, contributors to help with the project.

Regards,
Juan

**Comment by DorkWorks** [comment]
Hi,

Just to let know that I just started collecting components for this project and follow your development.

Dirk

**Reply by jareff-g** [reply]
Hello Dirk,

Thanks for letting me know, good luck with the project!
I assume you already have the link to Torulf's site but in case you don't it is [here](https://tscann8.torulf.com/index.html). 

Regards,
Juan

**Reply by jareff-g** [reply]
By the way, I just remembered I have a few PCBs laying around (as the minimum order is 5). If you have not ordered them yet and are interested, I can share one with you for the price I paid (proportional to 1 PCB, around 3 €) plus shipping. Let me know.

**Reply by DorkWorks** [reply]
Yes, I found Torulf's site some time ago. For the pcb, I think I will use a prototype pcb like Ton did. That way I can patch and change without much hassle.

I have spent a lot of time with electronics hobby in the past till life became busy, just retired and picking up the pieces. The attic is still full with boxes of old projects. I am now restoring an old digital clock with a bunch of TTL74xx ic's which I build as a teenager early 1973.

I already worked with Raspberries, Arduino is new so I have to study on that.

Dirk

**Reply by jareff-g** [reply]
Nice. I'm more dedicated to programming aspects. I had some interest in electronics when I was young (before the hobby computers era) but never got beyond putting together some commercial kits. Now I'm trying to make a version of the scanner electronics with a Raspberry P Pico replacing the Arduino Nano, we'll see how it ends.
Rgds/Juan

**Reply by jkytomaki** [reply]
Hey,

Reporting in from Helsinki, Finland! I've been building the Tscann 8 project for past 6 months or so. I've finally started running some test scans. 

I'm glad to have found Juan's work, both the modded parts in Thingiverse, the software forks and this discussion thread. I've been missing a community to ask for help and hints while building the scanner.

Some random questions, if anyone could help:

Do I understand it correctly that the original Arduino Nano should be replaced by an RPi Pico to use any of this repo's code, even just the UI bits? The Readme.md still has some references to Arduino, maybe this switch out could be mentioned in more detail.

I'm concerned about scratching the film with the film wheels or the film gate. Do you sand/buff your 3d print parts to get rid of any sharp corners or imperfections?

The official site says this about dust removal wheels:

> Apply double-sided tape to the wheels, then the back of electrical tape preferably of low quality so the film does not stick. Postit tape probably works well too.

Are any of you following this design? It seems risky to expose the film to the sticky side of electrical tape ("low quality" or not). If not, are you using tape non-sticky side up, or just replacing these wheels with maybe two film control wheels instead?

Thanks!

Janne

**Reply by jareff-g** [reply]
On the Arduino vs Pico topic: The code in this repo works with Torulf standard board, using an Arduino Nano. 

The Pico version is a personal project where I'm trying to have a new PCB for the scanner using a Pico instead of the Arduino. There was no initial objective when I started, besides the personal challenge (I don't have electronics background), but along the way I added a feature or two I thought could be useful. If I can share the details, if anyone is interested.

About the double sided tape, I had the same concern, and for a while I was using the film control rollers as you suggest. But then I made a few tests with some non-critical film , and saw it was not as bad as I thought, so started using it (and it actually works!).

**Reply by jkytomaki** [reply]
Thanks for your quick reply!

This bit in the readme.md added to the confusion:

>Always use both ALT-Scann8 modules (UI + Arduino) toghether, do not mix with T-Scann8 original software. At this point in time (October 2023) the interface between both has changed a lot and is no longer compatible.

I read this meaning that there is an ALT-Scann8 specific version of the Arduino code in the repo, but the only INO file I found was for the RPi Pico. So I should just leave the standard Torulf Arduino code on the Arduinon nano, and use your code from the master branch for the Python UI code?

**Reply by jareff-g** [reply]
That comment was because, when I started ALT-Scann8, I tried to keep the same interface (between the two modules, RPi and Arduino) as the original software from Torulf. But at one point it became too complex to maintain such compatibility, so I dropped it. So either you use the two modules from Torulf software, or you use both modules from this repo. Make sure not to mix them.

And yes, there is one INO file from the Arduino, it is ['ALT-Scann8-Controller.ino'](https://github.com/jareff-g/ALT-Scann8/blob/master/ALT-Scann8-Controller.ino). The INO for the Pico contains 'pico' in the filename just to differentiate it.

**Reply by jkytomaki** [reply]
Not sure how I missed the Arduino INO file! I even looked through the file listing twice because I expected it should be there. I guess I need to check my glasses prescription! 😅 Sorry for the confusion and thank you, will test these out and report back!

**Comment by DorkWorks** [comment]
Still waiting for the last components to arrive. In the meantime I captured the schematic for learning and to understand the working. Then, just for fun I decided to how difficult it would be to design a pcb. Not to difficult and I ordered five pieces at JCLPCB. Some small changes were made like connector for an IR Photo Interrupter for the Traction Stop and software switching the Power LED.

Next is looking at the STL files. Imho all wheels should be fitted with ball bearing.

![image](https://github.com/jareff-g/ALT-Scann8/assets/140604124/bbe7f502-88b1-4d1e-8b3a-aaf1def22f6b)

**Reply by jareff-g** [reply]
Funny, looks like you took the same couple of decisions I also took for my Pico board: replace the full 40 pin RPi connector by a smaller one, and control the backlight via software. I also thought about using those same connectors, but since I do not have the corresponding cramping tool I decided to stick with the Dupont ones.

Also, could you share some details about the traction stop using IR? 

![image](https://github.com/jareff-g/ALT-Scann8/assets/98735354/d96e2ac8-b9bf-4a8f-a787-0a38f842ecda)

**Reply by DorkWorks** [reply]
I have some Arduino KY-010 Photo Interrupter Modules, think I have to modify a little by moving the connector to the bottom. The stepper cables I bought were already fitted with JST connectors so I bought an assortment box with JST-XH connectors in various sizes and a crimping tool.

I must admit that till recently I only knew the Arduino by name and the Pico since a week. Are there advantages by using the Pico or does the Arduino have shortcomings for this kind of project?

**Reply by jareff-g** [reply]
That's interesting. I bought a similar sensor ([here](https://fr.aliexpress.com/item/1005004659912013.html?spm=a2g0o.order_list.order_list_main.28.22b85e5bYc5chK&gatewayAdapt=glo2fra)), but with the idea to add it somewhere in the rewind path, to be able to automatically stop the motor when rewind (or FF) is over. I have not tried it yet.
In the Arduino vs Pico topic you have plenty of videos in YouTube explaining the differences (mainly, being a more modern board, is has better specs in almost every aspect). However I didn't select it for anything specific, just because I had previous experience with it and wanted to see if I could do it.
